<?xml version="1.0"?>
<launch>
  <arg name="config_package" default="$(optenv ROBOT_NAVIGATION_CONFIG_PACKAGE summit_xl_navigation)"/>
  <arg name="config_folder" default="config"/>
  <arg name="config_path" default="$(eval find(config_package) + '/' + config_folder)"/>

  <!-- NOTE: NO namespace here - this is a global planner -->
  
  <!-- Topics arguments -->
  <arg name="cmd_vel_topic" default="move_base/cmd_vel"/>
  <arg name="odom_topic" default="robot/robotnik_base_control/odom"/>
  <arg name="front_scan_topic" default="robot/front_laser/scan"/>
  <arg name="rear_scan_topic" default="robot/rear_laser/scan"/>
  
  <!-- Frames arguments - use global frame names -->
  <arg name="global_frame" default="robot_map"/>
  <arg name="odom_frame" default="robot_odom"/>
  <arg name="base_frame" default="robot_base_footprint"/>

  <arg name="kinematics" default="diff"/>
  <arg name="footprint_model" default="$(optenv ROBOT_FOOTPRINT_MODEL rbkairos)"/>

  <!-- Run move_base at GLOBAL level (no namespace) -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <!-- Remap to combined costmap -->
    <remap from="map" to="/combined_costmap" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)" />
    <remap from="odom" to="$(arg odom_topic)" />
    
    <!-- Frame settings -->
    <param name="global_frame" value="$(arg global_frame)"/>
    <param name="base_frame" value="$(arg base_frame)"/>
    <param name="odom_frame" value="$(arg odom_frame)"/>
    
    <!-- Tell static_map_layer to use combined_costmap -->
    <param name="global_costmap/static_map_layer/map_topic" value="/combined_costmap" />
    
    <!-- Load configuration files -->
    <rosparam file="$(arg config_path)/move_base_params.yaml" command="load" />
    <rosparam file="$(arg config_path)/global_planner_params.yaml" command="load" />
    <rosparam file="$(arg config_path)/costmap_common_params.yaml" command="load" ns="global_costmap" subst_value="true"/>
    <rosparam file="$(arg config_path)/global_costmap_params_map.yaml" command="load" ns="global_costmap" subst_value="true"/>
    <rosparam file="$(arg config_path)/costmap_common_params.yaml" command="load" ns="local_costmap" subst_value="true"/>
    <rosparam file="$(arg config_path)/local_costmap_params.yaml" command="load" ns="local_costmap" subst_value="true"/>
    <rosparam file="$(arg config_path)/teb_local_planner_$(arg kinematics)_params.yaml" command="load" subst_value="true"/>
    <rosparam file="$(arg config_path)/planner_footprints/$(arg footprint_model).yaml" command="load"/>
    
    <!-- Override frames AFTER loading YAML files -->
    <param name="TebLocalPlannerROS/map_frame" value="$(arg global_frame)"/>
    <param name="global_costmap/global_frame" value="$(arg global_frame)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame)"/>
  </node>

</launch>